src/main/java/org/cbio/gdcpipeline/reader/ClinicalReader.java

```diff
@@ -56,30 +58,35 @@ public ClinicalDataModel read() throws Exception {
     @Override
     public void open(ExecutionContext executionContext) throws ItemStreamException {
         List<File> clinicalFileNames = getClinicalFileList();
-
-        if (clinicalFileNames.isEmpty()) {
-            throw new ItemStreamException("Empty Clinical Files list");
-        }
-        for (File clinicalFile : clinicalFileNames) {
-            try {
-                TcgaBcr tcgaBcr = unmarshall(clinicalFile);
-                addData(tcgaBcr);
-            } catch (JAXBException e) {
-                if (LOG.isErrorEnabled()) {
-                    LOG.error("Unmarshalling error. Skipping file " + clinicalFile.getName());
+        if (!clinicalFileNames.isEmpty()) {
+            for (File clinicalFile : clinicalFileNames) {
+                if (LOG.isInfoEnabled()) {
+                    LOG.info("Processing Clinical file : " + clinicalFile.getName());
                 }
-                continue;
-            }
-            if (filter_sample_flag.equalsIgnoreCase("true")) {
-                for (int m = 0; m < this.clinicalDataModelList.size(); m++) {
-                    ClinicalDataModel c = this.clinicalDataModelList.get(m);
-                    if (c instanceof Sample) {
-                        if (((Sample) c).getSample_id().endsWith("-10")) {
-                            clinicalDataModelList.remove(m);
+                try {
+                    TcgaBcr tcgaBcr = unmarshall(clinicalFile);
+                    addData(tcgaBcr);
+                } catch (JAXBException e) {
+                    if (LOG.isErrorEnabled()) {
+                        LOG.error("Unmarshalling error. Skipping file " + clinicalFile.getName());
+                    }
+                    continue;
+                }
+                if (filter_sample_flag.equalsIgnoreCase("true")) {
```

Comment:
Is there a reason to not do this filtering before you call the addData method in the try block as you loop through the files? Would save you an iteration through the sample list.